.github/workflows/code-analysis.yml

```diff
@@ -0,0 +1,41 @@
+name: Slither Analysis
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
```

Comment:
Instead of a separate file that also requires downloading foundry etc. why not just fold this into the primary ci workflow?

Reply:
The primary CI workflow is building also the test files (for tests) and the libraries and all. If we run slither on that it gives +200 vulnerability related to our tests and the libraries we're using.
However, in this separate workflow, it is only building the src contracts and no test files and running the code analysis only against them. So we don't have to go through 200 vulnerability and trying to find ones related to our source code.
If we want to use the primary CI workflow, that would mean we would need to delete the built files then rebuild the project different and that gets messy. I think this way is cleaner